src/map/utils/synthutils.cpp

```diff
@@ -835,6 +836,29 @@ int32 startSynth(CCharEntity* PChar)
 int32 doSynthResult(CCharEntity* PChar)
 {
     uint8 m_synthResult = PChar->CraftContainer->getQuantity(0);
+    if (map_config.anticheat_enabled)
+    {
+        std::chrono::duration animationDuration = server_clock::now() - PChar->m_LastSynthTime;
+        if (animationDuration < 5s)
+        {
+            // Attempted cheating - Did not spend enough time doing the synth animation,
+            // Synth therefore always fails.
+            #ifdef _DSP_SYNTH_DEBUG_MESSAGES_
+            ShowDebug(CL_CYAN"Caught player cheating by injecting synth done packet.\n");
+            #endif
+            PChar->CraftContainer->setQuantity(0, synthutils::SYNTHESIS_FAIL);
+            m_synthResult = SYNTHESIS_FAIL;
+            doSynthFail(PChar);
+            // And report the incident (will possibly jail the player)
+            anticheat::ReportCheatIncident(PChar,
+                anticheat::CHEAT_ID_FASTSYNTH,
+                anticheat::CHEAT_SEVERITY_PROBABLE,
+                (uint32)std::chrono::duration_cast<std::chrono::milliseconds>(animationDuration).count(),
+                "Player attempted to bypass synth animation by injecting synth done packet.");
```

Comment:
Do we really need `CHEAT_ID` if we're passing a verbose string to the table? Or at least have it tied in some fashion to the ID.

Reply:
It's intended to allow statistics SQL queries, e.g. find all players who used a specific cheat to see how widespread it is. In this case I put a constant line in the (optional) description but maybe others will make more versatile use of it.

Reply:
Do we need the description column at all then? CHEAT_ID saying exactly what they were jailed for?

Reply:
Just to give maximum flexibility. Yes, right now I used a constant string but it doesn't have to be that way. You can provide whatever information you want.

Reply:
I don't feel like we need both, if you want verbosity like that for every jail event stick with just the description column and scrap `CHEAT_ID`'s. You can just as easily query for a specific (or fuzzy) string as you can a number.

Reply:
I've accepted @ibm2431's proposal and will refactor this tomorrow to include a cheat_types table that allows server admins to configure the punitive measures for each one. Cheat ID will then serve as the primary key for that table. Let's continue this discussion tomorrow after I push the change.